modules/core/src/main/java/org/apache/ignite/internal/processors/cache/tree/AbstractPendingEntryInnerIO.java

```diff
@@ -42,7 +42,12 @@
         assert row.link != 0;
         assert row.expireTime != 0;
 
-        PageUtils.putLong(pageAddr, off, row.expireTime);
+        long expireTime = row.expireTime;
+
+        if (row.tombstone)
+            expireTime |= 0x8000000000000000L;
+
+        PageUtils.putLong(pageAddr, off, expireTime);
```

Comment:
storeByOffset & getLookupRow are identical for AbstractPendingEntryInnerIO and AbstractPendingEntryLeafIO. Should we move this code somewhere up the class hierarchy?

Reply:
This looks like a refactoring job. Let's fix it in another ticket.